test-api/api.ts

```diff
@@ -163,24 +163,26 @@ import * as api from '../src';
     subscribe: async function* () {
       yield true;
     },
+    resolve: (p) => p,
   });
 
   t.subscriptionField({
     name: 'foo',
     type: t.Boolean,
-    // @ts-expect-error: subscribe must return number not object with number property
     subscribe: async function* () {
       yield { foo: true };
     },
+    resolve: (p) => p.foo,
```

Comment:
Here the `resolve` function can also be optional as the default subscription field resolver is `(value) => value[fieldName]`.

Can we cover that with a test?

Reply:
In this case it's a different thing, isn't it? The type of this according to `type: t.Boolean` is
```
extend type Subscription {
  foo: Boolean
}
```
that `subscribe` returns an object that has an attribute called `foo` is just incidental.

Isn't this the test that's needed?
```
  t.subscriptionField({
    name: 'foo',
    type: t.Boolean,
    subscribe: async function* () {
      yield true;
    },
    // Here resolve should be allowed to be optional, 
    // since what subscribe returns extends t.Boolean
  });
```

Reply:
I updated the subscription tests. Lemme know if I misunderstood what you asked for

Reply:
This test ofc fails at the moment
```
  // subscribe yields a type that extends Out
  // so resolve can be omitted
  t.subscriptionField({
    name: 'foo',
    type: t.Boolean,
    subscribe: async function* () {
      yield true;
    },
  });
```